dist/fedora-infra/Dockerfile

```diff
@@ -0,0 +1,26 @@
+FROM fedora:30
+
+ADD . /src
+
+# build: system utilities and libraries
+RUN dnf -y install g++ openssl-devel
+
+# build: pinned Rust stable toolchain (only use in case of temporary distro breakage)
+#ADD https://static.rust-lang.org/dist/rust-1.35.0-x86_64-unknown-linux-gnu.tar.gz rust.tar.gz
+#RUN tar -xf rust.tar.gz --strip 1 && ./install.sh
+
+# build: system Rust toolchain
+RUN dnf -y install rust cargo
+
+# build: graph-builder binary
+RUN cd /src && cargo build --release --bin graph-builder && mv /src/target/release/graph-builder /usr/local/bin/
+
+# build: policy-engine binary
+RUN cd /src && cargo build --release --bin policy-engine && mv /src/target/release/policy-engine /usr/local/bin/
+
+# build: cleanup
+RUN cd /src && cargo clean
+
+# run: default config
+WORKDIR /
+CMD [ "/usr/local/bin/policy-engine" ]
```

Comment:
I would suggest using a multi-stage build to separate the target binaries from the build sources and toolchain.

Reply:
Unfortunately the environment where this runs does not support multi-stage. I plan to investigate [S2I chained-builds](https://docs.openshift.com/container-platform/3.11/dev_guide/builds/advanced_build_operations.html#dev-guide-chaining-builds) later on, and switch to them if positive.

Reply:
:man_shrugging: